### PR TITLE
19924/16661: Mhr Product Tile Content

### PR DESCRIPTION
**Pull request by cameron-eyds**
*Issue #:*
https://github.com/bcgov/entity/issues/19924
https://github.com/bcgov/entity/issues/16661

*Description of changes:*
* Coming Soon label for content NOT releasing with Qualified Supplier initial release
* Generic product 'note' implementation and mhr product note content
* typo fix
* Minor logic refactor to product find() to prioritize active and pending over rejected, in the event of multiple product requests.

Screenshot 2024-02-22 at 6 38 10 AM

Screenshot 2024-02-22 at 6 38 21 AM

Screenshot 2024-02-22 at 6 38 29 AM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2738-p5642b6w.web.app

**Review by dimak1 — APPROVED** [review]

File: auth-web/src/components/auth/common/Product.vue [hunk 1]
```diff
@@ -157,6 +157,11 @@
                 class="mb-0"
                 v-html="$t(productLabel.details)"
               />
+              <p
+                v-if="$te(productLabel.note)"
+                class="mb-0"
+                v-html="$t(productLabel.note)"
```
Comment (seeker25) [inline auth-web/src/components/auth/common/Product.vue:163]: please use `v-sanitize`

File: auth-web/src/components/auth/common/Product.vue [hunk 2]
```diff
@@ -244,6 +249,7 @@ export default class Product extends Mixins(AccountMixin) {
     let { code } = this.productDetails
     let subTitle = `${code && code.toLowerCase()}CodeSubtitle`
     let details = `${code && code.toLowerCase()}CodeDescription`
+    let note = `${code && code.toLowerCase()}CodeNote`
```
Comment (seeker25) [inline auth-web/src/components/auth/common/Product.vue:252]: would `code?.toLowerCase()` work?

**Review by seeker25 — APPROVED** [review]

File: auth-web/src/components/auth/common/Product.vue (hunk 1 above)
Comment (cameron-eyds) [reply auth-web/src/components/auth/common/Product.vue:163]: Can do, would you like this for the previously existing v-html's here, as well?

File: auth-web/src/components/auth/common/Product.vue (hunk 2 above)
Comment (cameron-eyds) [reply auth-web/src/components/auth/common/Product.vue:252]: Absolutely! will update this and the others 👍

File: auth-web/src/components/auth/common/Product.vue (hunk 1 above)
Comment (seeker25) [reply auth-web/src/components/auth/common/Product.vue:163]: yup, sounds good

**Comment by seeker25** [comment]
Before we merge this, let me check with John to figure out the timeline for removing most of the text regarding BCEID

**Comment by cameron-eyds** [comment]
> Before we merge this, let me check with John to figure out the timeline for removing most of the text regarding BCEID, I have a feeling that could be done in a day and it might need to go out ASAP

Sounds good, no rush at all!

File: auth-web/src/components/auth/common/Product.vue (hunk 2 above)
Comment (cameron-eyds) [reply auth-web/src/components/auth/common/Product.vue]: Cleans up a lot of code, optionalChaining 🥳 
Short circuit used to be the way before ts brought this in, its the little things in life ya know!

File: auth-web/src/components/auth/common/Product.vue (hunk 1 above)
Comment (cameron-eyds) [reply auth-web/src/components/auth/common/Product.vue]: Updated!
One thing to note for future reference, classes assigned in the template string don't translate through the sanitize. Simple thing to change to a bold tag vs a class, just something noticed 👍 

Screenshot 2024-02-22 at 9 10 04 AM

**Review by doug-lovett — APPROVED** [review]
Looks good.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2738-p5642b6w.web.app

**Comment by cameron-eyds** [comment]
Screenshot 2024-02-25 at 4 45 01 PM


@seeker25 Is this still pending, just looking at getting some eyes on these content changes before we release this upcoming week :) 
No rush, just trying to give Divya some lead-time 👍

**Comment by seeker25** [comment]
We have some changes but nothing crazy

**Comment by cameron-eyds** [comment]
> We have some changes but nothing crazy, if you give me access to your auth repo, I'll handle the package version updates

Sent you an inv!

**Pull request merged**